patroni/postgresql/config.py

```diff
@@ -972,12 +955,25 @@ def resolve_connection_addresses(self) -> None:
 
         tcp_local_address = {'host': tcp_local_address, 'port': port}
 
-        self._local_address = unix_local_address if self._config.get('use_unix_socket') else tcp_local_address
         self.local_replication_address = unix_local_address\
             if self._config.get('use_unix_socket_repl') else tcp_local_address
 
         self._postgresql.connection_string = uri('postgres', netloc, self._postgresql.database)
-        self._postgresql.set_connection_kwargs(self.local_connect_kwargs)
+
+        local_address = unix_local_address if self._config.get('use_unix_socket') else tcp_local_address
```

Comment:
I feel like this needs a comment to explain the intention here. Then that makes me think that this method is doing a number of different steps to build up different data points. There's 2 separate places where data is changed outside of this method. Should there be a single place where this updates? Also do we envisage a possibility of the below failing and if so do the previous data write(s) need rolling back or not be set?